dockstore-client/src/main/java/io/dockstore/client/cli/nested/ToolClient.java

```diff
@@ -714,6 +705,25 @@ public File downloadTargetEntry(String toolpath, ToolDescriptor.TypeEnum type, b
         }
     }
 
+    /**
+     * Returns the version ID for the provided entry path
+     * @param entryPath Tool path
+     */
+    @Override
+    public String getVersionID(String entryPath) {
+        final String[] parts = entryPath.split(":");
+
+        String versionID = parts.length > 1 ? parts[1] : null;
+
+        if (versionID == null) {
+            final DockstoreTool container = getDockstoreTool(parts[0]);
+            versionID = container.getDefaultVersion();
+        }
+
+        // last resort use 'latest' tag
+        return versionID == null ? "latest" : versionID;
```

Comment:
What if latest doesn't exist?

Reply:
[Updated](https://github.com/dockstore/dockstore-cli/blob/2a2f2ea14f509729cf60b324bfd05beb55fac418/dockstore-client/src/main/java/io/dockstore/client/cli/nested/ToolClient.java#L714). This now more closely resembles the implementation in the WorkflowClient class, while still conforming to the original standard from [here](https://github.com/dockstore/dockstore-cli/blob/e05578d5a95d3f57757939b89ee280fb53a717a1/dockstore-client/src/main/java/io/dockstore/client/cli/nested/ToolClient.java#L680). Similar to that in WorkflowClient, If there is no latest/default version, we will default to the version that was built most recently.